target/linux/ath79/image/lzma-loader/src/loader.c

```diff
@@ -243,7 +285,27 @@ void loader_main(unsigned long reg_a0, unsigned long reg_a1,
 		default:
 			printf("unknown error %d!\n", res);
 		}
+#if (LZMA_WRAPPER)
 		halt();
+#else
+		if (CONFIG_FLASH_ALT != 0) {
```

Comment:
We know at compile time whether or not there is an alternative location at all. No need to compile for single-image scenarios.

Reply:
True, the only difference would be the size of the loader...you can decide whether that is important enough